joinmarket-webui/jam
# fix: display error message if backend is unreachable

Resolves #538.

Changeset is based on #521 (please review/merge #521 first).

Before this PR the app showed a generic error message when the backend was unreachable.
After this PR the app displays an error message with more informative reason, e.g. "No connection to backend: Gateway Timeout.".

At first I tried implementing this behaviour via the recommended `react-router-dom`'s `errorElement` property (as [described in the docs](https://reactrouter.com/en/main/start/tutorial#not-found-data)). However, this did not work for reasons not yet known. I will try it again with a future release, as this is a more appropriate error handling mechanism and also enables better separation of data and views, which would be nice to have.

#### How to test
Checkout the `master` branch and start the app (with `npm start`) without starting the regtest environment - verify that you see a generic error message. Switch to this branch (`git checkout error-page`) and verify that you see a more informative message.

### :camera_flash: Before/After
<img src="https://user-images.githubusercontent.com/3358649/194900400-431d9618-5029-49a0-ae65-457d7c57bb6b.png" width=400 /><img src="https://user-images.githubusercontent.com/3358649/194900308-8bbe9406-5e8f-4af6-995d-7c30924f9a3a.png" width=400 />
